hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/KeyValueHandler.java

```diff
@@ -942,9 +941,14 @@ public Container importContainer(ContainerData originalContainerData,
       final InputStream rawContainerStream,
       final TarContainerPacker packer)
       throws IOException {
+    Preconditions.checkState(originalContainerData instanceof
+        KeyValueContainerData, "Should be KeyValueContainerData instance");
 
     KeyValueContainerData containerData =
         new KeyValueContainerData(originalContainerData);
+    // schemaVersion should be used to decide the container db path
+    containerData.setSchemaVersion(
+        ((KeyValueContainerData) originalContainerData).getSchemaVersion());
```

Comment:
Move this schemaVersion set into KeyValueContainerData(ContainerData data) constructor, change the constructor parameter type to KeyValueContainerData instead.

Reply:
Hmmm, I'll try to add schemaVersion to the constructor.